src/Config/Converters/NestedCreateOptionsConverter.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.DataApiBuilder.Config.ObjectModel;
+
+namespace Azure.DataApiBuilder.Config.Converters
+{
+    /// <summary>
+    /// Converter for the nested create operation options.
+    /// </summary>
+    internal class NestedCreateOptionsConverter : JsonConverter<NestedCreateOptions>
+    {
+        /// <inheritdoc/>
+        public override NestedCreateOptions? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            if (reader.TokenType == JsonTokenType.Null)
+            {
+                return null;
+            }
+
+            if (reader.TokenType is JsonTokenType.StartObject)
+            {
+                NestedCreateOptions? nestedCreateOptions = null;
+                while (reader.Read())
+                {
+                    if (reader.TokenType == JsonTokenType.EndObject)
+                    {
+                        break;
+                    }
+
+                    string? propertyName = reader.GetString();
```

Comment:
what if propertyName is null here?

Reply:
Added a null check and an exception is thrown when `propertyName` turns out to be `null`